clang/include/clang/Serialization/ASTReader.h

```diff
@@ -942,6 +942,12 @@ class ASTReader
   /// Sema tracks these to emit deferred diags.
   llvm::SmallSetVector<serialization::DeclID, 4> DeclsToCheckForDeferredDiags;
 
+  /// The module files imported by different module files. Indirectly imported
+  /// module files are included too. The information comes from
+  /// ReadModuleOffsetMap(ModuleFile&).
+  mutable llvm::DenseMap<ModuleFile *, llvm::SmallVector<ModuleFile *>>
+      ImportedModuleFiles;
```

Comment:
Why does this live in `ASTReader`? If we moved it to `ModuleFile`, we could remove `mutable` and the map lookups.

Reply:
Done in the new commit.

There was a similar member `Imports` in `ModuleFile`, which records the directly imported module files from the coding. But, yes, it is good to remove the map lookups and let's try to handle the ambiguous by commenting.